gitlab/protocols/http_backend.py

```diff
@@ -0,0 +1,29 @@
+import sys
+from abc import abstractmethod
```

Comment:
I personally (I think @nejch may have a different opinion) am not a fan of importing methods/classes.

I think just `import abc` is good and then easier when reading the code below to know where it comes from.

This is also recommended by Google's Python Style Guide, though I am not suggesting we adopt Google's Python Style Guide 🙂 And yes I realize that we are breaking this guideline in our current code 😥

https://google.github.io/styleguide/pyguide.html

2.2 Imports

Use import statements for packages and modules only, not for individual classes or functions.